lib/pf/web/dispatcher.pm

```diff
@@ -59,6 +62,13 @@ Reference: http://perl.apache.org/docs/2.0/user/handlers/http.html#PerlTransHand
 =cut
 
 sub handler {
+    alarm $portal_request_timeout;
+    my $res = _handler(@_);
+    alarm 0;
+    return $res;
+}
```

Comment:
This does not follow the usual alarm pattern in perl.
See the example here for setting an alarm in mod_perl.
https://perl.apache.org/docs/2.0/user/coding/coding.html#Using_Signal_Handlers